internal/resultstore/resultstore.go

```diff
@@ -56,6 +60,58 @@ func (rs *ResultStore) generatePath(p Pkg) string {
 	return path
 }
 
+func (rs *ResultStore) SaveWriteBufferZip(ctx context.Context, p Pkg, fileName string, writeBufferPaths []string) error {
+	bkt, err := blob.OpenBucket(ctx, rs.bucket)
+	if err != nil {
+		return err
+	}
+	defer bkt.Close()
+
+	uploadPath := filepath.Join(rs.generatePath(p), fileName+".zip")
+	log.Info("Uploading results",
+		"bucket", rs.bucket,
+		"path", uploadPath)
+
+	w, err := bkt.NewWriter(ctx, uploadPath, nil)
+	defer w.Close()
+	if err != nil {
+		return err
+	}
+
+	zipWriter := zip.NewWriter(w)
+	defer zipWriter.Close()
+
+	for _, filePath := range writeBufferPaths {
+		file, err := os.OpenFile(filePath, os.O_RDWR, 0666)
+		if err != nil {
+			return err
+		}
+		fileContents, err := os.ReadFile(filePath)
+		if err != nil {
+			return err
+		}
+		writeBufferId := utils.GetSHA256Hash(fileContents)
+		w, err := zipWriter.Create(writeBufferId)
+		if err != nil {
+			return err
+		}
+
+		if _, err := io.Copy(w, file); err != nil {
+			return err
+		}
+
+		if err = file.Close(); err != nil {
+			return err
+		}
```

Comment:
This could be written more efficiently:

```go
file, err := os.Open(filePath)
if err != nil {
//...
}
defer file.Close() // This is in-case there are errors below, calling it twice generates an error, but we don't check it.

h := sha256.New()

if _, err := io.Copy(file, h); err != nil {
//...
}

fileHash := hex.EncodeToString(hash.Sum(nil))

w, err := zipWriter.Create(fileHash)
if err != nil {
//...
}

if _, err := file.Seek(0,0); err != nil {
//...
}

if _, err := io.Copy(w, file); err != nil {
//...
}

if err := file.Close(); err != nil {
//...
}

Reply:
Does copy require us to seek back to the beginning of the file? It was hard to tell from the documentation.